### ISSUE TITLE
Handle state params in oauthGoogleEventHandler

### ISSUE THREAD

**Issue by julianfox** [atinux/nuxt-auth-utils#178]
It seems that the oauthGoogleEventHandler doesn't support passing a state parameter during the OAuth flow ?

Here more information from the docs: 
https://developers.google.com/identity/protocols/oauth2/web-server#request-parameter-state

**Comment by julianfox** [comment]
I did a fork that should works, but I didn't test yet:
https://github.com/julianfox/nuxt-auth-utils/blob/add-state-to-google-oauth/src/runtime/server/lib/oauth/google.ts

**Comment by julianfox** [comment]
I confirm that my fork works as intended.

**To integrate it into my code:**
I copied and pasted my version of oauth/google.ts into server/utils/oauth/google.ts, renaming OAuthGoogleConfig to OauthGoogleStateConfig and oauthGoogleEventHandler to oauthGoogleStateEventHandler.
I also had to copy and paste server/lib/utils.ts into the server/utils/oauth folder.

Is there a better way to extend handlers in our code?
